nupic/research/frameworks/dendrites/mixins/eval_per_task.py

```diff
@@ -0,0 +1,101 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+__all__ = [
+    "EvalPerTask",
+]
+
+
+class EvalPerTask:
+    """
+    Mixin for getting metrics, namely accuracy, for each task independently
+    """
+
+    def validate(self, loader=None):
+        """
+        Run validation on the currently active tasks, and gather accuracy measures
+        for each of those tasks independently.
+        """
+
+        # Figure this out
+        if loader is None:
+            loader = self.val_loader
+
+        results = {}
+        for task in range(self.current_task + 1):
+            self.val_loader.sampler.set_active_tasks(task)
+
+            # Store results for each individual task in results[task]
+            task_results = super().validate(loader)
+            results[task] = task_results
+
+        # Loop over results per task to get averages
+        task_average_results = self.average_over_tasks(results)
+        results["task_average"] = task_average_results
+
+        # Flatten dictionary and add task id to key name (e.g. mean_accuracy_4)
+        results = self.format_results(results)
+
+        self.val_loader.sampler.set_active_tasks(self.current_task)
+
+        return results
+
+    def format_results(self, results):
+        """
+        Flatten results dictionary and add task id to key name
+
```

Comment:
`format_results` and `average_over_tasks` do not reference `self`, and therefore should be module-level functions, not class methods.